### PR TITLE
fix: image layout

### PR DESCRIPTION
**Pull request by matijakljajic**

### CODE REVIEW

File: src/lib/components/search/load/images/LoadImage.svelte [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <script lang="ts">
 	// types
 	import type { Width } from '$lib/types/width';
+	import { randomWidth } from '$lib/types/width';
```
Comment (aleksasiriski) [inline src/lib/components/search/load/images/LoadImage.svelte:4]: This isn't used anymore, why?

File: src/lib/components/search/display/images/Preview.svelte [hunk 2]
```diff
@@ -8,12 +8,16 @@
...
-			class="mx-auto h-[50vh] w-auto hover:ring-2 hover:ring-hearchco-primary hover:dark:ring-hearchco-secondary object-cover transform transition duration-200 ease-in-out"
-		/>
+		<div
+			class="h-[50dvh] flex justify-center mx-auto hover:ring-2 hover:ring-hearchco-primary hover:dark:ring-hearchco-secondary overflow-hidden rounded-lg"
+		>
+			<img
+				id="link-{result.Rank}"
+				src={result.ImageResult.ThumbnailURL}
+				alt={result.Title}
+				class="h-full w-full object-contain transform transition duration-200 ease-in-out"
```
Comment (aleksasiriski) [inline src/lib/components/search/display/images/Preview.svelte:18]: What's the duration and ease-in-out used here? There's no animation happening.

File: src/lib/components/search/display/images/Preview.svelte [hunk 3]
```diff
@@ -8,12 +8,16 @@
...
 <div id="image-preview-{result.Rank}" class="px-4 pt-4 pb-10 w-full lg:sticky lg:top-0">
 	<a href={result.URL}>
-		<img
-			id="link-{result.Rank}"
-			src={result.ImageResult.ThumbnailURL}
-			alt={result.Title}
-			class="mx-auto h-[50vh] w-auto hover:ring-2 hover:ring-hearchco-primary hover:dark:ring-hearchco-secondary object-cover transform transition duration-200 ease-in-out"
-		/>
+		<div
+			class="h-[50dvh] flex justify-center mx-auto hover:ring-2 hover:ring-hearchco-primary hover:dark:ring-hearchco-secondary overflow-hidden rounded-lg"
```
Comment (aleksasiriski) [inline src/lib/components/search/display/images/Preview.svelte:12]: Why remove duration and ease-in-out for the hover?

File: src/lib/components/search/display/images/Images.svelte [hunk 4]
```diff
@@ -22,31 +22,27 @@
...
-				<Image {result} bind:imgResultPreview />
-			</div>
-		{/each}
-	</section>
+	<div class="w-full">
+		<section id="images" class="grid grid-cols-fit auto-rows-[200px] grid-flow-dense gap-2">
+			{#each results as result (result.URL)}
+				<div
+					class:row-span-2={(result.ImageResult.Original.Width * 0.8) /
+						result.ImageResult.Original.Height >
```
Comment (aleksasiriski) [inline src/lib/components/search/display/images/Images.svelte:30]: Switch to using thumbnail for calculating aspect ratio, since that is what is being used as source in the grid.

File: src/lib/components/search/load/images/LoadImages.svelte [hunk 5]
```diff
@@ -2,22 +2,15 @@
...
-	<section id="images" class="flex flex-wrap justify-center gap-2">
+	<section id="images" class="grid grid-cols-fit auto-rows-[200px] grid-flow-dense gap-2">
 		{#each { length: numberOfResults } as _, i}
 			<div class="flex-none">
-				{#if i % 3 === 0 && i % Math.floor(2 * Math.random()) !== 0}
-					<LoadImage colored={true} width={randomWidth()} />
-				{:else}
-					<LoadImage width={randomWidth()} />
-				{/if}
+				<LoadImage colored={Math.random() <= 0.1} />
```
Comment (aleksasiriski) [inline src/lib/components/search/load/images/LoadImages.svelte:13]: This doesn't spread the colored images anymore, why the change?

File: src/lib/components/search/display/images/Preview.svelte (hunk 3 above)
Comment (matijakljajic) [reply src/lib/components/search/display/images/Preview.svelte:12]: Good find, copy-paste mistake.

File: src/lib/components/search/display/images/Images.svelte (hunk 4 above)
Comment (matijakljajic) [reply src/lib/components/search/display/images/Images.svelte:30]: As I've said, aspect ratio is a _proportional relationship_.
I'll change it so that the code-base is more consistent.
Google Images issue that we're dealing with is not related to this, I'm certain.

File: src/lib/components/search/load/images/LoadImages.svelte (hunk 5 above)
Comment (matijakljajic) [reply src/lib/components/search/load/images/LoadImages.svelte:13]: Because this kind of randomness aesthetically worked only with the previous layout solution.
I backtracked and remembered your suggestion regarding this. The simpler, the better, as you've said.

File: src/lib/components/search/display/images/Images.svelte (hunk 4 above)
Comment (matijakljajic) [reply src/lib/components/search/display/images/Images.svelte]: _declared as a bug which was in turn solved so this can be resolved_
